### PR TITLE
openstack: implement basic metadata client functionality

### PR DESCRIPTION
**Pull request by katexochen**
### Additional info

- There are still things missing to fulfill all interfaces.

### Checklist

- [x] Add labels (e.g., for changelog category)

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -238,8 +238,14 @@ type OpenStackConfig struct {
 	//   Floating IP pool to use for the VMs. For details see: https://docs.openstack.org/ocata/user-guide/cli-manage-ip-addresses.html
 	FloatingIPPoolID string `yaml:"floatingIPPoolID" validate:"required"`
 	// description: |
-	//   Service account token to use authenticate VMs with the OpenStack API. Alternatively leave empty and pass value via CONSTELL_OPENSTACK_SERVICE_ACCOUNT_TOKEN environment variable.
-	ServiceAccountToken string `yaml:"serviceAccountToken" validate:"required"`
+	//   UserDomainName is the name of the domain where a user resides.
+	UserDomainName string `yaml:"userDomainName" validate:"required"`
+	// description: |
+	// 	 Username to use inside the cluster.
```
Comment (daniel-weisse) [inline internal/config/config.go:244]:
```suggestion
	//   UserDomainName is the name of the domain where a user resides.
	UserDomainName string `yaml:"userDomainName" validate:"required"`
	// description: |
	//   Username to use inside the cluster.
```
white space

File: internal/cloud/openstack/openstack.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+					continue
+				}
+
+				if addr.IPVersion != ipV4 {
+					continue
+				}
+
+				if addr.IP == "" {
+					continue
+				}
```
Comment (daniel-weisse) [inline internal/cloud/openstack/openstack.go:175]: I would write this as one If statement, can also be kept as is if you prefer it this way

Comment (katexochen) [reply internal/cloud/openstack/openstack.go:175]: I prefer it this way, much faster to read imo.

**Comment by katexochen** [comment]
:floppy_disk: Image to test: 

```
ref/feat-cloud-openstack-metadata/stream/debug/v2.6.0-pre.0.20230307122230-4f5ec3477b73
```

**Review by daniel-weisse — APPROVED** [review]
Didn't test on openstack, but code lgtm

**Review by malt3 — APPROVED** [review]
lgtm

**Pull request merged**